src/Tuura/Plato/Translate/Translation.hs

```diff
@@ -93,3 +93,27 @@ toTransitions = map (\l -> Transition (variable l) (polarity l))
 --Create a tuple containing a list of possible causes, for each effect.
 arcLists :: [Causality (Transition a)] -> [([Transition a], Transition a)]
 arcLists xs = [ (f, t) | Causality f t <- xs ]
+
+convert :: Enum a => CircuitConcept a -> CircuitConcept Signal
+convert c = mempty
+         {
+           initial = convertInitialValue (initial c),
+           arcs = fmap convertCausality (arcs c),
+           interface = convertInterface (interface c),
+           invariant = fmap convertInvariant (invariant c)
+         }
+
+convertInitialValue :: Enum a => (a -> InitialValue) -> (Signal -> InitialValue)
+convertInitialValue f (Signal i) = f $ toEnum i
```

Comment:
I guess this can be generalised to type `Enum a => (a -> b) -> (Signal -> b)` without changing the implementation? Then you don't need to duplicate code in `convertInterface`.